packages/tools/devtools/devtools-example/src/test/ui-smoke-test/ChromeBrowserDevtool.test.ts

```diff
@@ -0,0 +1,56 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+/* eslint-disable spaced-comment */
+/// <reference types="jest-environment-puppeteer" />
+/// <reference types="puppeteer" />
+/// <reference types="jest" />
+
+import { retryWithEventualValue } from "@fluidframework/test-utils";
+import { globals } from "../../../jest.config";
+
+jest.setTimeout(30_000);
+describe("End to end tests", () => {
+	/**
+	 * Gets the value of the text form backed by our CollaborativeTextArea.
+	 *
+	 * @remarks Assumes there is only one `textarea` element on the page.
+	 *
+	 * @param expectedValue - The value we expect the value of the text area to be.
+	 */
+	async function getTextFormValue(expectedValue: string): Promise<string> {
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-return
+		return retryWithEventualValue(
+			/* callback: */ async () =>
+				page.evaluate(() => {
+					const divs = document.querySelectorAll(".textarea");
+					const textAreaElements = divs[0].querySelectorAll("textarea");
+					const textarea = textAreaElements[0];
+					return textarea?.value;
+				}),
+			/* check: */ (actualValue) => actualValue === expectedValue,
+			/* defaultValue: */ "",
+		);
+	}
+
+	beforeAll(async () => {
+		// Wait for the page to load first before running any tests
+		// so this time isn't attributed to the first test
+		await page.goto(globals.PATH, { waitUntil: "load", timeout: 0 });
+	}, 45_000);
+
+	beforeEach(async () => {
+		await page.goto(globals.PATH, { waitUntil: "load" });
+		// eslint-disable-next-line @typescript-eslint/no-unsafe-return
+		await page.waitForFunction(() => globalThis.fluidStarted);
+		await page.waitForSelector("textarea");
+	});
+
+	it("Smoke: verify test app can be launched", async () => {
+		// Verify by checking for text area associated with the SharedString.
+		const textArea = await getTextFormValue("Enter text here.");
+		expect(textArea).toEqual("Enter text here.");
```

Comment:
The way the test works feels a bit wonky... shouldn't we just be doing `page.waitForSelector()` on an id we give to the element we care about, and then check that the text in it is what we expect? Here the helper `getTextFromValue` seems like it would only ever return the expected value.

Reply:
Right! Adding wait for to ensure app view is fully rendered.